src/components/Mapbox.tsx

```diff
@@ -0,0 +1,130 @@
+import { Result, useSearchState } from '@yext/search-headless-react';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import ReactDOM from 'react-dom';
+
+/**
+ * Interface for customizing the {@link Mapbox} map derived from Mapbox API's Map options.
+ * The "container" field is set internally. The "center" field is offered as a surface level prop
+ * in {@link MapboxProps}.
+ *
+ * @public
+ */
+export type MapboxCustomOptions = Omit<MapboxOptions, 'container' | 'center'>;
+
+/**
+ * Interface for customizing the {@link Mapbox} marker derived from Mapbox API's Marker options.
+ * The "element" field is set internally.
+ *
+ * @public
+ */
+export type MarkerCustomOptions = Omit<MarkerOptions, 'element'>;
+
+/**
+ * An outline to create a custom Marker component.
+ *
+ * @public
+ */
+export type MapPinComponent<TEntity> = (props: { result: Result<TEntity> }) => JSX.Element;
+
+/**
+ * Props for the {@link Mapbox} component.
+ *
+ * @public
+ */
+export interface MapboxProps<TEntity> {
+  mapboxApiKey: string,
+  mapboxOptions?: MapboxCustomOptions,
+  markerOptions?: MarkerCustomOptions,
+  PinComponent?: MapPinComponent<TEntity>,
+  defaultCenter?: Coordinate,
+  getCoordinate: CoordinateGetter<TEntity>
+}
+
+/**
+ * A component that renders a map with markers to show result locations.
+ *
+ * @param props - {@link MapboxProps}
+ * @returns A React element conatining a Google Map
+ *
+ * @public
+ */
+export function Mapbox<TEntity>({
+  mapboxApiKey,
+  mapboxOptions,
+  markerOptions,
+  PinComponent,
+  defaultCenter: center,
+  getCoordinate
+}: MapboxProps<TEntity>) {
+  mapboxgl.accessToken = mapboxApiKey;
+  const mapContainer = useRef(null);
+  const map = useRef<Map | null>(null);
+  const markers = useRef<Marker[]>([]);
+
+  const locationResults = useSearchState(state => state.vertical.results) as Result<TEntity>[];
+  const hasLocationResults = locationResults && locationResults.length !== 0;
+
+  const mapboxStyle = mapboxOptions?.style ?? 'mapbox://styles/mapbox/streets-v11';
+
+  useEffect(() => {
+    if (mapContainer.current && !map.current) {
+      const options: MapboxOptions = {
+        container: mapContainer.current,
+        style: mapboxStyle,
+        center: center ? [center.longitude, center.latitude] : undefined,
+        ...mapboxOptions
+      };
+      map.current = new Map(options);
+    }
+    map.current?.resize();
+  });
+
+  useEffect(() => {
+    for (const marker of markers.current) {
+      marker.remove();
+    }
+    const bounds = new LngLatBounds();
+    markers.current = [];
+
+    if (map.current && hasLocationResults) {
+      for (const result of locationResults) {
+        const markerLocation = getCoordinate(result);
+        if (markerLocation) {
+          const { latitude, longitude } = markerLocation;
+          const el = document.createElement('div');
+          PinComponent && ReactDOM.render(<PinComponent result={result} />, el);
```

Comment:
why are you using ReactDOM.render here?

Reply:
This was one of the things that I wanted to go back and revisit-- I took this small 2 line snippet from davish's example since he implemented it in a way that we can use imported pin components-- Looks like renders the pin to a div, then we insert that div into the Map.  Is there a better way to do that?  I'll of course look around.

Reply:
I didn't find a good way to do this as Mapbox's marker option takes in HTMLElement only for the DOM element to use as a marker, so this needs to be pre-rendered before passing into new Marker()